docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1081 @@
+              "speaker": "CLARENCE",
+              "text_entry": "To make a bloody supper in the Tower."
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+The `suggest` parameter finds suggestions using only prefix matching.
+For example, you don't get back "To be, or not to be," which you might want as a suggestion.
+To work around this issue, manually add curated suggestions and add weights to prioritize your suggestions.
+
+Index a document with an input suggestion and assign a weight:
+
+```json
+PUT shakespeare/_doc/1
+{
+  "text": "To m",
+  "text_entry": {
+    "input": [
+      "To be, or not to be: that is the question:"
+    ],
+    "weight": 10
+  }
+}
+```
+
+Perform the same search as before:
+
+```json
+GET shakespeare/_search
+{
+  "suggest": {
+    "autocomplete": {
+      "prefix": "To m",
+      "completion": {
+        "field": "text_entry",
+        "size": 3
+      }
+    }
+  }
+}
+```
+
+You see the indexed document as the first result:
+
+```json
+{
+  "took": 1021,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 0,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "suggest": {
+    "autocomplete": [
+      {
+        "text": "To m",
+        "offset": 0,
+        "length": 5,
+        "options": [
+          {
+            "text": "To be, or not to be: that is the question:",
+            "_index": "shakespeare",
+            "_type": "_doc",
+            "_id": "1",
+            "_score": 30,
+            "_source": {
+              "text": "To me",
+              "text_entry": {
+                "input": [
+                  "To be, or not to be: that is the question:"
+                ],
+                "weight": 10
+              }
+            }
+          },
+          {
+            "text": "To make a bastard and a slave of me!",
+            "_index": "shakespeare",
+            "_type": "_doc",
+            "_id": "5369",
+            "_score": 4,
+            "_source": {
+              "type": "line",
+              "line_id": 5370,
+              "play_name": "Henry VI Part 1",
+              "speech_number": 2,
+              "line_number": "4.5.15",
+              "speaker": "JOHN TALBOT",
+              "text_entry": "To make a bastard and a slave of me!"
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+Use the `term` suggester to suggest corrected spellings for individual words.
+The `term` suggester uses an edit distance to compute suggestions. Edit distance is the number of characters that need to be changed for a term to match.
+
+If a user misspells a search term:
+
+```json
+GET shakespeare/_search
+{
+  "suggest": {
+    "spell-check": {
+      "text": "lief",
+      "term": {
+        "field": "text_entry"
+      }
+    }
+  }
+}
+```
+
+Use the `term` suggester to return a list of corrections:
```

Comment:
The `term` suggester returns a list of corrections:

Reply:
Done